### PR TITLE
Make available_on optional

### PR DESCRIPTION
**Pull request by maful**
Issue https://github.com/joemasilotti/railsdevs.io/issues/46#issuecomment-962522548

image

### CODE REVIEW

File: db/migrate/20211107024057_change_developers_available_on.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ChangeDevelopersAvailableOn < ActiveRecord::Migration[7.0]
+  def change
+    change_column_null :developers, :available_on, true
```
Comment (joemasilotti) [inline db/migrate/20211107024057_change_developers_available_on.rb:3]: Oh! I didn't know this was a thing. TIL.

Comment (maful) [reply db/migrate/20211107024057_change_developers_available_on.rb:3]: I just found out too 

API https://edgeapi.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-change_column_null

**Review by joemasilotti — APPROVED** [review]
Looks great! Thanks for the PR :)

File: db/migrate/20211107024057_change_developers_available_on.rb (hunk 1 above)
Comment (joemasilotti) [reply db/migrate/20211107024057_change_developers_available_on.rb:3]: I've always done `change_column :developers, :available_on, null: false` but you have to remember to copy over any other params, like `default:`. Yours is so much better!

**Pull request merged**